operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -307,6 +311,86 @@ func (r *TenantReconciler) checkValidWorkspaces(ctx context.Context, tn *crownla
 	return tenantExistingWorkspaces, workspaces, err
 }
 
+// deleteClusterNamespace deletes the namespace for the tenant, if it fails then it returns an error.
+func (r *TenantReconciler) deleteClusterNamespace(ctx context.Context, tn *crownlabsv1alpha2.Tenant, nsName string) (err error) {
+	ns := v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: nsName}}
+
+	nsErr := utils.EnforceObjectAbsence(ctx, r.Client, &ns, "personal namespace")
+
+	if nsErr != nil {
+		klog.Errorf("Error when deleting namespace of tenant %s -> %s", tn.Name, nsErr)
+	}
+
+	return nsErr
+}
+
+// enforceNamespaceKeepAliveOrDelete deletes the namespace if it should be deleted.
+func (r *TenantReconciler) enforceNamespaceKeepAliveOrDelete(ctx context.Context, tn *crownlabsv1alpha2.Tenant, nsName string, tenantWorkspaceKeepAlive time.Duration) (keepNsOpen bool, err error) {
```

Comment:
tenantWorkspaceKeepAlive does not need to be passed as parameter, as part of the receiver.

Reply:
@giorio94 I don't understand this comment, wouldn't the parameter tenantWorkspaceKeepAlive be necessary in order to do the comparison in enforceNamespaceKeepAliveOrDelete?

Reply:
I will close this conversation because I believe `tenantWorkspaceKeepAlive` is necessary

Reply:
No, it is not. Just access it as r.TenantWorkspaceKeepAlive from the function, and drop the parameter.